connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java

```diff
@@ -252,7 +252,13 @@ private static ConfigDef defineCheckpointConfig(ConfigDef baseConfig) {
                         ConfigDef.Type.CLASS,
                         TOPIC_FILTER_CLASS_DEFAULT,
                         ConfigDef.Importance.LOW,
-                        TOPIC_FILTER_CLASS_DOC);
+                        TOPIC_FILTER_CLASS_DOC)
+                .define(
```

Comment:
This connector is not emitting offset-syncs so I'm not sure it makes sense to define this configuration on this connector.

I see you use this for validation, but in standalone and distributed modes the value for this configuration on this connector is not necessarily the same as the value on MirrorSourceConnector. So if you start MirrorSourceConnector with `emit.offset-syncs.enabled=false` and then this connector with the value set to `true` it would still pass validation.

When running in dedicated mode the configurations are shared, but maybe we can do the validation in the MirrorMakerConfig class.

Reply:
How would move the validation to `MirrorMakerConfig` make different? Am actually re-thinking about this validation now and I am leaning a bit toward removing it as it might be okay to not validate `MirrorCheckpointConnector` and treat it in same way as if we are deploying a client before creating the topic. The connector will just do nothing

Reply:
I'm not sure I folly understand your question. My main point is that we should not define the `emit.offset-syncs.enable` configuration on the checkpoint connector as it has no functional impact and a very limited use in validating the configuration because the source connector could use a different value.

Reply:
My comment about `MirrorMakerConfig` is that in dedicated mode, the checkpoints and source connector share the same `emit.offset-syncs.enable` value, so in this case it can be use to effectively validate the config.

Reply:
I moved this to `MirrorMakerConfig`